File: modules/agent/src/mocks/controllers/orderController.ts
import { Router, Request, Response } from 'express';
import * as orderService from '../services/orderService.js';

export const orderRouter = Router();

interface OrderItem {
  sku: string;
  qty: number;
  unit_price: number;
  location: string;
}

interface CreateOrderRequest {
  customer_id: string;
  items: OrderItem[];
}

interface UpdateStatusRequest {
  status: string;
}

orderRouter.get('/orders', (req: Request, res: Response) => {
  const status = req.query.status as string | undefined;
  const customerId = req.query.customer_id as string | undefined;
  res.json(orderService.getOrders(status, customerId));
});

orderRouter.get('/orders/summary', (_req: Request, res: Response) => {
  res.json(orderService.getOrderSummary());
});

orderRouter.get('/orders/:orderId', (req: Request, res: Response) => {
  const order = orderService.getOrderDetail(req.params.orderId);
  if (!order) {
    res.json({ error: `Order ${req.params.orderId} not found` });
    return;
  }
  res.json(order);
});

orderRouter.post('/orders', (req: Request, res: Response) => {
  const body = req.body as CreateOrderRequest;
  res.json(orderService.createOrder(body.customer_id, body.items));
});

orderRouter.patch('/orders/:orderId/status', (req: Request, res: Response) => {
  const body = req.body as UpdateStatusRequest;
  res.json(orderService.updateOrderStatus(req.params.orderId, body.status));
});

orderRouter.get('/customers', (_req: Request, res: Response) => {
  res.json(orderService.getCustomers());
});

File: modules/agent_deprecated/src/agent/mocks/controllers/order_controller.py
from typing import Optional

from fastapi import APIRouter, Query
from pydantic import BaseModel

from agent.mocks.services import order_service

router = APIRouter(prefix="/api", tags=["orders"])


class OrderItem(BaseModel):
    sku: str
    qty: int
    unit_price: float
    location: str


class CreateOrderRequest(BaseModel):
    customer_id: str
    items: list[OrderItem]


class UpdateStatusRequest(BaseModel):
    status: str


@router.get("/orders")
async def get_orders(
    status: Optional[str] = Query(None),
    customer_id: Optional[str] = Query(None),
):
    return order_service.get_orders(status, customer_id)


@router.get("/orders/summary")
async def get_order_summary():
    return order_service.get_order_summary()


@router.get("/orders/{order_id}")
async def get_order(order_id: str):
    order = order_service.get_order_detail(order_id)
    if not order:
        return {"error": f"Order {order_id} not found"}
    return order


@router.post("/orders")
async def create_order(req: CreateOrderRequest):
    items = [item.model_dump() for item in req.items]
    return order_service.create_order(req.customer_id, items)


@router.patch("/orders/{order_id}/status")
async def update_order_status(order_id: str, req: UpdateStatusRequest):
    return order_service.update_order_status(order_id, req.status)


@router.get("/customers")
async def get_customers():
    return order_service.get_customers()
